src/org/thoughtcrime/securesms/ConversationListFragment.java

```diff
@@ -153,21 +149,28 @@ public void onActivityCreated(Bundle bundle) {
 
     setHasOptionsMenu(true);
     initializeFabClickListener(false);
-    initializeListAdapter();
+    list.setAdapter(new ConversationListAdapter(getActivity(), GlideApp.with(this), locale, this));
+    loadChatlist();
+    chatlistJustLoaded = true;
   }
 
   @Override
   public void onResume() {
     super.onResume();
 
     updateReminders();
-    list.getAdapter().notifyDataSetChanged();
+
+    if (getActivity().getIntent().getIntExtra(RELOAD_LIST, 0) == 1
+        && !chatlistJustLoaded) {
+      loadChatlist();
+    }
+    chatlistJustLoaded = false;
```

Comment:
shouldn't `chatlistJustLoaded = true;` if the previous `loadChatlist();` got executed?